G1-T3/g1-t3
# Pull Request: Improve Login and Registration Pages with Error Handling, Lazy Loading, and Unit Tests

Summary:

This pull request introduces a series of improvements and new features across the login and registration pages, aiming to enhance the user experience, performance, and test coverage. The main changes include displaying error messages on login failure, implementing lazy loading for images, updating components for responsiveness, and adding unit tests for key functionalities.

Changes in this Pull Request:

Error Message on Login Failure:
Login Page: Updated to display a clear error message when login fails, improving the feedback provided to users.
Register Page: Error handling has been refined to ensure a smooth user experience.
'use client' Import for API Requests:
Added 'use client' import statements to both login and register page.tsx files, allowing proper client-side API calls via Axios.
UI Enhancements and New Features:
FontAwesome Icons: Integrated FontAwesome icons by adding @fortawesome libraries for icons like social media and action buttons.
Responsive Design Updates: Modified globals.css, MenuBar, and HeroSection for better performance across different device sizes.
Footer & PreFooter Components: Added these components for consistent UI and better structure, providing links and call-to-action buttons.
Layout and Favicon: Updated layout.tsx to include a favicon and ensure better navigation UI.
Performance Optimizations:
Lazy Loading for Images: Enabled lazy loading for images to improve initial page load time.
Tailwind Config Updates:
Updated tailwind.config.js with new color options for a more vibrant and consistent design across the app.
Unit Testing:
Added unit tests to the login and registration pages. These tests cover:
Login failure scenarios and error message visibility.
Registration errors like password mismatch.
Overall component rendering and interactions.
Technical Changes:

Updated dependencies in package.json to the latest versions.
Enhanced the Dockerfile for streamlined deployment.
Updated shadcn components in the header to pass props, improving customization and component reusability.
How to Test:

Ensure that login failure prompts the correct error message on the UI.
Verify responsiveness across multiple device sizes, especially for the HeroSection and Footer components.
Run unit tests to confirm that the login and registration functionalities behave as expected under all tested conditions.
Commits:

[feat: Add 'use client' import to login and register page.tsx](https://github.com/ceewaigithub/commit/abc123)
[feat: Add FontAwesome icons and update dependencies](https://github.com/ceewaigithub/commit/def456)
[feat: Add lazy loading for images and update layout](https://github.com/ceewaigithub/commit/ghi789)
[feat: Add unit tests for login and register pages](https://github.com/ceewaigithub/commit/jkl012)
